internal/sqlutil/sql.go

```diff
@@ -107,3 +111,40 @@ func SQLiteDriverName() string {
 	}
 	return "sqlite3"
 }
+
+func minOfInts(a, b int) int {
+	if a <= b {
+		return a
+	}
+	return b
+}
+
+type QueryProvider interface {
+	QueryContext(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error)
+}
+
+const SQLite3MaxVariables = 999
+
+//RunLimitedVariablesQuery split up a query with more varaibles than the used databse can handle in multiple queries.
+func RunLimitedVariablesQuery(ctx context.Context, query string, qp QueryProvider, rowHandler func(*sql.Rows) error, variables []interface{}) error {
+	var start int
+	for start < len(variables) {
+		n := minOfInts(len(variables)-start, SQLite3MaxVariables)
+		query := strings.Replace(query, "($1)", QueryVariadic(n), 1)
+		rows, err := qp.QueryContext(ctx, query, variables[start:start+n]...)
+		if err != nil {
+			return err
+		}
+		err = rowHandler(rows)
+		if err := rows.Close(); err != nil {
```

Comment:
This can't be called `err` else it will replace the `err` from `rowHandler` without checking it.

Reply:
Actually, I don't think that is the case. The err from rows.Close() is in a new scope inside the if (https://play.golang.org/p/tnQq8KHKVwn), but for clarity it might be better to use another name.